### PR TITLE
fix(List): placement and outline shape

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
The List outline weird shape was fixed, but a regression was added with the automatic dropup/dropdown.
capture d ecran 2018-08-06 a 13 52 33

The reason: we need dropdown menus to be invisible by default, rendered and displayed, so we can get its position to decide if we have an overflow.
If it is displayed, and takes its place, the outline is weird.

**What is the chosen solution to this problem?**
* Set back the non rendering of the menu (display: none)
* In dropdown, keep its open state and on component update, we add the `dropup` classname if needed

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js [hunk 1]
```diff
@@ -134,36 +127,42 @@ class ActionDropdown extends React.Component {
...
+			const dropdownMenu = dropdownTrigger.nextSibling;
+			const dropdownContainer = getDropdownContainer(dropdownTrigger);
 
-			this.setState(oldState => {
-				if (!oldState.dropup && dropdownRect.bottom > containerRect.bottom) {
-					return { dropup: true };
-				} else if (oldState.dropup && dropdownRect.top < containerRect.top) {
-					return { dropup: false };
+			if (dropdownContainer) {
+				console.log('container found', dropdownContainer.tagName);
```
Comment (sgangnant) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: Did you leave those on purpose ? Plus, it adds error to the linting report (above)

Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: Oh no, thanks

Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: + I'll suppress the other linter errors as I haven't found any other way to do that

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js [hunk 2]
```diff
@@ -110,79 +111,72 @@ describe('InjectDropdownMenuItem', () => {
...
+		const isDropup = true;
+		const isDropdown = false;
 
-		const wrapper = shallow(<ActionDropdown items={[{ label: 'item 1' }, { label: 'item 2' }]} />);
-		expect(wrapper.state().dropup).toBeFalsy();
+		function testSwitch({
+			containerPosition,
+			menuPosition,
+			isInitialDropup,
+			isDropupOrIsDropdown,
```
Comment (jdescottes-talend) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: Either the naming or the data stored is weird. I don't understand what's the meaning. When seeing it in standalone, It seems to be always true, and when seeing with the usage, I just don't understand neither ^^'

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js [hunk 3]
```diff
@@ -134,36 +128,40 @@ class ActionDropdown extends React.Component {
 	constructor(props) {
 		super(props);
 		this.onToggle = this.onToggle.bind(this);
-		this.state = {
-			dropup: props.dropup,
-		};
+		this.state = {};
```
[the remark this answers is not in the document]
Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: By default a dropdown is not opened. So isOpen is falsy.

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js [hunk 4]
```diff
@@ -110,79 +111,67 @@ describe('InjectDropdownMenuItem', () => {
...
-		// dropdown that is under the bottom of container
-		const bottomOverflowEvent = getActionDropdownEvent(20, 40, divContainerElement);
-		// dropdown that is under the bottom of body element
-		const bodyBottomOverflowEvent = getActionDropdownEvent(20, 40, bodyElement);
+		const isDropup = true;
+		const isDropdown = false;
 
-		const wrapper = shallow(<ActionDropdown items={[{ label: 'item 1' }, { label: 'item 2' }]} />);
-		expect(wrapper.state().dropup).toBeFalsy();
+		function testSwitch({ containerPosition, menuPosition, isInitialDropup, expectedDropupState }) {
```
Comment (jdescottes-talend) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: If I'm not mislead, this naming matching the data held : `isDropupExpected`

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js [hunk 5]
```diff
@@ -110,79 +111,67 @@ describe('InjectDropdownMenuItem', () => {
...
+			// given
+			const container = document.createElement('div');
```
Comment (jdescottes-talend) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: When checking the tests in local, I can't find a way to go through this cases function.

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js [hunk 6]
```diff
@@ -110,79 +111,67 @@ describe('InjectDropdownMenuItem', () => {
...
-		// when / then
-		wrapper.props().onToggle(true, bodyBottomOverflowEvent);
-		expect(wrapper.state().dropup).toBe(true);
+		cases('dropdown/dropup switch', testSwitch, [
+			{
+				name: 'should dropup on dropdown bottom overflow',
+				containerPosition: { top: 0, bottom: 35 },
+				menuPosition: { top: 20, bottom: 40 },
+				isInitialDropup: false,
+				isDropupOrIsDropdown: isDropup,
```
Comment (jdescottes-talend) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: The props cases need to be updated too.
And the data corresponding can be directly given ; `isDropup` and `isDropdown`, even if I see now why they are, are not obvious, and misleading. Simple boolean without variables would avoid thinking :p

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js (hunk 4 above)
Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: done

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js (hunk 5 above)
Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js]: fixed

File: packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js [hunk 7]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable-next-line import/no-extraneous-dependencies */
```
Comment (romainseb) [inline packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: Could it be better to add react-dom in peer-dependency of @talend/react-component as we use it this way ?

Comment (jsomsanith) [reply packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js]: You're totally right, I made the change, thanks

**Pull request merged**